packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -170,6 +173,18 @@ class InnerBlocks extends Component {
 			/>
 		);
 
+		// Wrap context provider if (and only if) block has context to provide.
+		const blockType = getBlockType( block.name );
+		if ( blockType && blockType.providesContext ) {
```

Comment:
It's more useful than that but I like the way you phrased it :)

Reply:
> This could be a once in a lifetime opportunity to use `?.`.

Aha! I'm always trying to find a use for it. I'm surprised I missed this, but you make a good point, it seems appropriate here.

Reply:
> This could be a once in a lifetime opportunity to use `?.`.

Addressed in b06177b48.

Reply:
> This could be a once in a lifetime opportunity to use `?.`.

I had to revert this in 2aa56cad6. It was causing errors from the automated documentation generator:

```
⇒ npm run docs:build

> gutenberg@7.9.1 docs:build /Users/andrew/Documents/Code/gutenberg
> node ./docs/tool/index.js && node ./bin/api-docs/update-api-docs.js

(node:14765) UnhandledPromiseRejectionWarning: Error: Command failed with exit code 1: /Users/andrew/Documents/Code/gutenberg/node_modules/.bin/docgen packages/block-editor/src/index.js --output packages/block-editor/README.md --to-token --use-token "Autogenerated API docs" --ignore "/unstable|experimental/i"

SyntaxError: Unexpected token .

    at makeError (/Users/andrew/Documents/Code/gutenberg/node_modules/execa/lib/error.js:58:11)
    at handlePromise (/Users/andrew/Documents/Code/gutenberg/node_modules/execa/index.js:114:26)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Transform.<anonymous> (/Users/andrew/Documents/Code/gutenberg/bin/api-docs/update-api-docs.js:206:4)
(node:14765) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:14765) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Related: #19952

I'm pretty alarmed it was never caught in continuous integration 🤔 It should be run almost exactly the same way:

https://github.com/WordPress/gutenberg/blob/886bb15ed0bd2de9046ad1fcec76528b4b5637c0/.travis.yml#L124-L132

https://github.com/WordPress/gutenberg/blob/886bb15ed0bd2de9046ad1fcec76528b4b5637c0/package.json#L195

Reply:
Oh goodness!

https://travis-ci.com/github/WordPress/gutenberg/jobs/320538572

The error occurs, _**and**_ the build still passes 😱 

I'll create a follow-up task.

Reply:
> I'll create a follow-up task.

I made a pull request instead: #21690